Automattic/wp-calypso
# Share Site: Remove `onSuccess` from useQuery to upgrade to v5

- Related to https://github.com/Automattic/wp-calypso/pull/70242#discussion_r1407483562

## Proposed Changes

* Keep the same functionality just a refactor on useQuery to avoid using the native React Query `onSuccess`.

## Testing Instructions



* On an atomic site that is not launched, go to `/settings/general/$siteSlug`
* Find the Launch site card
* Try the share site toggle
* Verify the toggle works as expected. Same behaviour as before.

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/b397d46c-d2b5-4f89-871b-a65e13fc50ba

Update toggle value if has been changed on another tab.


https://github.com/Automattic/wp-calypso/assets/779993/b2e67ba1-37cd-4a64-a7eb-11660e09a939



## Pre-merge Checklist
